src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double lateralSide;
+    private double upperBase;
+    private double bottomBase;
+    private double height;
+    private double triangleLeg;
+
+    public IsoscelesTrapezoid() {
+    }
+
+    public IsoscelesTrapezoid(double upperBase, double bottomBase, double lateralSide) {
+        this.typeOfFigure = TypeOfFigure.ISOSCELES_TRAPEZOID;
+        this.area = calculateArea(upperBase, bottomBase, triangleLeg, height);
+        this.upperBase = upperBase;
+        this.bottomBase = bottomBase;
+        this.lateralSide = lateralSide;
+        this.triangleLeg = (bottomBase - upperBase) / 2;
+        this.height = getTrapezoidHigh(lateralSide, triangleLeg);
+    }
+
+    public double getTrapezoidHigh(double lateralSide, double triangleLeg) {
+        return Math.sqrt(Math.pow(lateralSide, 2) - Math.pow(triangleLeg, 2)); //Pythagorean theorem
+    }
+
+    public double calculateArea(double upperBase, double bottomBase,
```

Comment:
The `calculateArea` method should not have parameters since it can access the class fields directly. This will make the method signature simpler and the method easier to use.